### PR TITLE
fixes ptr to cstring warnings[backport]

### PR DESCRIPTION
**Pull request by ringabout**
ref https://github.com/nim-lang/Nim/pull/20761

Either one should be backported to Nim 1.6.10 rc2.

### CODE REVIEW

File: tests/stdlib/tnet_ll.nim [hunk 1]
```diff
@@ -29,7 +29,7 @@ suite "inet_ntop tests":
     check: ip4.s_addr == 0x10111213'u32
 
     var buff: array[0..255, char]
-    let r = inet_ntop(AF_INET, cast[pointer](ip4.s_addr.addr), buff[0].addr, buff.len.int32)
+    let r = inet_ntop(AF_INET, cast[pointer](ip4.s_addr.addr), cast[cstring](buff[0].addr), buff.len.int32)
     let res = if r == nil: "" else: $r
```
Comment (arnetheduck) [inline tests/stdlib/tnet_ll.nim:33]: `addr buffer` vs `addr buffer[0]`

Comment (arnetheduck) [reply tests/stdlib/tnet_ll.nim:33]: what's the (practical) difference?

Comment (ringabout) [reply tests/stdlib/tnet_ll.nim:33]: I don't know its practical difference. Pre-existing.

Comment (Araq) [reply tests/stdlib/tnet_ll.nim:33]: `addr buffer[0]` also works for strings and sequences.

**Comment by ringabout** [comment]
@narimiran backport please, thanks!

**Comment by narimiran** [comment]
> @narimiran backport please, thanks!

I just did.
There were several merge conflicts, mostly for files in `lib/std/private` that exist in `devel` but not in `version-1-6` — if the contents of those files were moved from somewhere else, that 'somewhere else' should probably get the same treatment.
Can you take a look?

**Comment by ringabout** [comment]
done => https://github.com/nim-lang/Nim/pull/20861

**Pull request merged**